test/e2e/common/node/checkpoint_container.go

```diff
@@ -0,0 +1,157 @@
+	var result restclient.Result
+	finished := make(chan struct{}, 1)
+	go func() {
+		result = c.CoreV1().RESTClient().Post().
+			Resource("nodes").
+			SubResource("proxy").
+			Name(fmt.Sprintf("%v:%v", node, port)).
+			Suffix(endpoint).
+			Do(context.TODO())
+
+		finished <- struct{}{}
+	}()
+	select {
+	case <-finished:
+		return result, nil
+	case <-time.After(proxyTimeout):
+		return restclient.Result{}, nil
+	}
+}
+
+var _ = SIGDescribe("Checkpoint Container [NodeFeature:CheckpointContainer]", func() {
+	f := framework.NewDefaultFramework("checkpoint-container-test")
+	var podClient *framework.PodClient
+	ginkgo.BeforeEach(func() {
+		podClient = f.PodClient()
+	})
+
+	ginkgo.It("will checkpoint a container out of a pod", func() {
+		ginkgo.By("creating a target pod")
+		pod := podClient.CreateSync(&v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{Name: "checkpoint-container-pod"},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:    "test-container-1",
+						Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+						Command: []string{"/bin/sleep"},
+						Args:    []string{"10000"},
+					},
+				},
+			},
+		})
+
+		p, err := podClient.Get(
+			context.TODO(),
+			pod.Name,
+			metav1.GetOptions{},
+		)
+
+		framework.ExpectNoError(err)
+		isReady, err := testutils.PodRunningReady(p)
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(
+			isReady,
+			true,
+			"pod should be ready",
+		)
+
+		framework.Logf(
+			"About to checkpoint container %q on %q",
+			pod.Spec.Containers[0].Name,
+			pod.Spec.NodeName,
+		)
+		result, err := proxyPostRequest(
+			f.ClientSet,
+			pod.Spec.NodeName,
+			fmt.Sprintf(
+				"checkpoint/%s/%s/%s",
+				f.Namespace.Name,
+				pod.Name,
+				pod.Spec.Containers[0].Name,
+			),
+			framework.KubeletPort,
+		)
+
+		framework.ExpectNoError(err)
+
+		err = result.Error()
+		if err != nil {
+			statusError, ok := err.(*apierrors.StatusError)
+			if !ok {
+				framework.Failf("got error %#v, expected StatusError", err)
+			}
+			// If we are testing against a kubelet with ContainerCheckpoint == false
+			// we should get a 404. So a 404 is (also) a good sign.
+			if (int(statusError.ErrStatus.Code)) == http.StatusNotFound {
+				ginkgo.By("Feature 'ContainerCheckpoint' is not enabled and not available")
+				return
+			}
+
+			// If the container engine has not implemented the Checkpoint CRI API
+			// we will get 500 and a message with
+			// '(rpc error: code = Unimplemented desc = unknown method CheckpointContainer'
+			if (int(statusError.ErrStatus.Code)) == http.StatusInternalServerError {
+				if strings.Contains(
+					statusError.ErrStatus.Message,
+					"(rpc error: code = Unimplemented desc = unknown method CheckpointContainer",
+				) {
+					ginkgo.By("Container engine does not implement 'CheckpointContainer'")
+					return
+				}
+			}
+			framework.Failf("Unexpected status code (%d) during 'CheckpointContainer'", statusError.ErrStatus.Code)
```

Comment:
This seems like a fair thing to test given that container runtimes can't yet support the new CRI.

Reply:
Not sure if this is just a comment or a request to change something. Are you asking for a change? It does not sound like it.

Reply:
No change, just a comment.